### PR TITLE
[5] add host database port

### PR DESCRIPTION
**Pull request by alikon**
Draft partial Pull Request for Feature request #43902 .

### Summary of Changes
add db host port filed

### Testing Instructions
ìnstall and play with different setup db hostname and db port

### Actual result BEFORE applying this Pull Request
postgres pdo doesn't work if port different from default (5432)

### Expected result AFTER applying this Pull Request
you can set a  db port 

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: installation/src/Helper/DatabaseHelper.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ abstract class DatabaseHelper
      *
      * @since   1.6
      */
-    public static function getDbo($driver, $host, $user, $password, $database, $prefix, $select = true, array $ssl = [])
+    public static function getDbo($driver, $host, $port, $user, $password, $database, $prefix, $select = true, array $ssl = [])
```
Comment (dgrammatiko) [inline installation/src/Helper/DatabaseHelper.php]: - move the port argument to the end
- use a default value (`$port = 3306`

This way you don't introduce a hard B/C break

**Comment by muhme** [comment]
Thank you @alikon for this draft PR, I added a language overwrite and did a picture:

shoot

 There are arguments for (e.g. using `2001:db8::1` and `1066`) and against an additional field for the port number (e.g. using `[2001:db8::1]:1066`). I vote for a new field because two reasons:
 1. The documentation. We have to give help on screen and to describe it in detail and this is easier with separate field and not using additional square brackets for IPv6 addresses
 2. Security checks with two separate fields for host and port number are easier to implement and maintain as with single field
 
Why do we need backwards compatibility here? And there is no hurry, from my point of view we can have it 5.2.

Another point: I would make the field optional and blank by default if the default database port is used. This again for simplicity: If we display the default port number, then we have to change it when the database is changed.

File: installation/src/Helper/DatabaseHelper.php (hunk 1 above)
Comment (alikon) [reply installation/src/Helper/DatabaseHelper.php]: done thx for the feedback

**Comment by degobbis** [comment]
I like the idea behind this PR, but I don't think it makes sense to make the field required.
If the field is empty, the default port should be used automatically.
In addition, it would certainly make sense to check the previous entry in “Host” to see whether a port was already specified there and, if necessary, to move it to the new field and to change the configuration.php.

**Comment by alikon** [comment]
with https://github.com/joomla-framework/database/pull/310 no more needed

in cypress.config.mjs  `db_host: 'localhost:5433'`

**Pull request closed without merging**